erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -633,6 +633,29 @@ def make_item_gl_entries(self, gl_entries):
 								"project": item.project or self.project
 							}, account_currency, item=item))
 
+						# check if the exchange rate has changed
+						purchase_receipt_conversion_rate = frappe.db.get_value('Purchase Receipt', {'name': item.purchase_receipt}, ['conversion_rate'])
+						if purchase_receipt_conversion_rate and self.conversion_rate != purchase_receipt_conversion_rate:
```

Comment:
Along with conversion rate diff we will also have to check if the net rate is same for both the item in both the docs

Reply:
Done.